### DISCUSSION TITLE
`0.0.14` - `regex` keyword on `Field` object not working (should be `pattern` in Pydantic v2)

### DISCUSSION THREAD

**Discussion by DarylStark** [fastapi/sqlmodel#735] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import Field, Relationship
from pydantic import Field as Field2

class User(SQLModel):
    fullname: str = Field(regex=r'^[A-Za-z0-9\- ]+$', max_length=128)
    fullname2: str = Field2(pattern=r'^[A-Za-z0-9\- ]+$', max_length=128)

fails = User(fullname='test', fullname2='test_failing') # Fails because it knows how to process the regex

succeeds = User(fullname='test_not_failing', fullname2='test') # Does not fail because the `regex` keyword is ignored
```

### Description

I'm migrating from SQLModel 0.0.8 to SQLModel 0.0.14 using Pydantic 2.5.2. In SQLModel 0.0.8 I could use the `regex` keyword from the `Field` class to create a regular expression validation, but that doesn't work anymore. According to the docs of Pydantic, this should be `pattern`, but the Field object of SQLModel doesn't support that named argument.

### Operating System

Linux

### Operating System Details

Pydantic version 2.5.2 (Core = 2.14.5)

### SQLModel Version

0.0.14

### Python Version

3.10.13

**Comment by MatMoore** [comment]
I can reproduce this issue as well.

As a workaround I was able to use `schema_extra` like this:

```python
class User(SQLModel):
    fullname: str = Field(max_length=128, schema_extra={'pattern': r'^[A-Za-z0-9\- ]+$'})
```

although I'm not sure if this is completely equivalent.

If the current version of sqlmodel is targeting both pydantic 1 and pydantic 2, perhaps SQLModel's `Field` should use `pattern` as well and support `regex` as an alias?

**Reply by dnszero** [reply]
Thank you! 

It turns out this is still the best solution in 2025 with SQLModel 0.0.24

**Comment by DarylStark** [comment]
I was unaware of the `schema_extra` argument. Thank you! That helps a lot!

**Comment by clembir** [comment]
It seems that the SQLModel Field class are not fully compatible with Pydantic v2 Field class.
- Some attributes are missing like `alias_priority`, `validation_alias`, `serialization_alias`,  `examples`, `json_schema_extra`, `frozen`, `validate_default`, `init_var`, `kw_only`, `strict`, `allow_inf_nan` and `union_mode`.

- Some attributes need to be updated like `regex` which is now `pattern` as explained above.

- Some attributes need to be deleted because they still refer to Pydantic v1 Field class like `comp`, `min_items`, `max_items`, `unique_items` and `allow_mutation`.

This is not a lot of work to make the modification in the code as it's just necessary to update the Field class in the `main.py` file with these attributes and the appropriate annotations.

**Comment by simonryf** [comment]
Linking the PR :-)
[https://github.com/fastapi/sqlmodel/pull/1231](https://github.com/fastapi/sqlmodel/pull/1231)

**Comment by ZougaRida** [comment]
As of this date, the problem of `regexp` still exists, as it should instead be named `pattern`.
